src/nodes.coffee

```diff
@@ -5169,6 +5200,14 @@ exports.For = class For extends While
     fragments.push @makeCode(returnResult) if returnResult
     fragments
 
+  ast: (o, level) ->
+    addToScope = (name) ->
```

Comment:
The loop variables of a `for` represent variable declarations (if they don't already exist)

This helper could probably be used to DRY up some of these other places where we're setting `isDeclaration` since it encapsulates the basic pattern of possibly adding to the scope (via `o.scope.find()`) and then setting `isDeclaration` based on whether it already existed in the scope